### ISSUE TITLE
SC2154 Doesn't Parse Function Names Containing Colons Correctly

### ISSUE THREAD

**Issue by stevekuznetsov** [koalaman/shellcheck#688]
The following code is incorrectly flagged with SC2154:
```sh
$ cat /tmp/exmaple.sh 
#!/bin/bash
function namespace::funcname() {
	echo 'Hello World!'
}
readonly -f namespace::funcname

$ shellcheck /tmp/exmaple.sh 

In /tmp/exmaple.sh line 5:
readonly -f namespace::funcname
            ^-- SC2154: namespace::funcname is referenced but not assigned.

```

**Comment by stevekuznetsov** [comment]
I guess it is possible the converse is the bug: the following example *isn't* flagged with SC2154:
```sh
$ cat /tmp/exmaple.sh 
#!/bin/bash
function func_name() {
	echo 'Hello World!'
}
readonly -f n func_name 

$ shellcheck /tmp/exmaple.sh
$
```

In either case, while it is true that the function is not called in the file in which it is declared, as the file is a utility library in my case, I would not expect 2154 to apply to function names.

**Comment by stevekuznetsov** [comment]
@koalaman any thoughts on this?

**Comment by koalaman** [comment]
This is indeed a bug, `readonly -f` shouldn't concern variables at all. It's been fixed in 3a7dc86. Thanks!

**Comment by stevekuznetsov** [comment]
@koalaman Thanks for the update!
